akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -754,14 +787,21 @@ object Source {
    *
    * @param bufferSize size of buffer in element count
    * @param overflowStrategy Strategy that is used when incoming elements cannot fit inside the buffer
-   * @param maxConcurrentOffers maximum number of pending offers when buffer is full, should be greater than 0
+   * @param maxConcurrentOffers maximum number of pending offers when buffer is full, should be greater than 0, not
+   *                            applicable when `OverflowStrategy.dropNew` is used
    */
   def queue[T](
       bufferSize: Int,
       overflowStrategy: OverflowStrategy,
       maxConcurrentOffers: Int): Source[T, SourceQueueWithComplete[T]] =
-    new Source(
-      scaladsl.Source.queue[T](bufferSize, overflowStrategy, maxConcurrentOffers).mapMaterializedValue(_.asJava))
+    overflowStrategy match {
+      case _: OverflowStrategies.DropNew => queue[T](bufferSize).mapMaterializedValue(_.asJavaSourceQueue())
```

Comment:
Are we sure that we want to change the queue implementation for DropNew? It's probably much better, but the change in behavior could impact users. I think we should consider deprecating this version of `Source.queue` instead and promote the new method instead. The deprecation could come in a separate PR (possibly later release)  since we probably want to provide a good alternative replacement for at least `Backpressure` strategy.

Reply:
I'll revert this change.

In the future I suggest we deprecate these two overloads altogether (with a backpressure and "fail" strategy alternative).  We can't deprecate `OverflowStrategies.DropNew` because it's used in `Source.actorRef` as well.

Reply:
and we might not need a fail strategy because that can be a decision for the caller (and implemented together with a KillSwitch)